RecipeApi/Authentication/SameAuthorRequirementHandler.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using RecipeApi.Database;
+
+namespace RecipeApi.Authentication
+{
+    public class SameAuthorRequirementHandler : AuthorizationHandler<SameAuthorRequirement, Recipe>
+    {
+        public SameAuthorRequirementHandler()
+        {
+        }
+
+        protected override Task HandleRequirementAsync(AuthorizationHandlerContext context,
+        SameAuthorRequirement requirement,
+        Recipe recipe)
+        {
+            var userId = context.User.FindFirst(c => c.Type == ClaimTypes.NameIdentifier).Value;
+
+            if (recipe.UserId == int.Parse(userId))
+            {
+                context.Succeed(requirement);
+            }
+
+            if (context.User.HasClaim(ClaimTypes.Role, "Admin"))
```

Comment:
It's better to use it in attributes together with roles.
https://learn.microsoft.com/en-us/aspnet/web-api/overview/security/authentication-and-authorization-in-aspnet-web-api
[Authorize(Roles=...)]

Reply:
To uodate something you must be author of this or admin. I check there id is equal to author id.

For admin it will be incorrect so I add check roles to accept also admin